### PR TITLE
[quest] ADDED: 'Alonso the Dragonslayer' Rune Quest To QuestDB

### LINKED ISSUE #5414: [quest] 'Alonso the Dragonslayer' Missing From QuestDB
## Description

The SoD quest 'Alonso the Dragonslayer' is not currently in the QuestDB. 

## Wowhead link

https://www.wowhead.com/classic/quest=78144/alonso-the-dragonslayer

### Discord Bug Report

Quest ID: 78144
Quest Name: Alonso the Dragonslayer
Quest Text: 
Objective Text: 
Reward Text: I've done it <playername>! At last my foe is defeated!

Oh the clash of fang and metal; of scale and blade. What peril, what thrill! And through it all, I arise a champion. 

Let us not forget your humble contribution as well! Without the armor you retrieved, I surely would not have bested the fiend. 

You have been an excellent squire, <playername>. Perhaps, one day, you too shall walk the glorious path of a knight.
Reward XP: 0
Questgiver: Creature-0-5175-1-32529-210995-000016203D
Character: Lvl 25 HUMAN WARRIOR
Player Coords:  [1440]  (42.026, 69.124)
QuestLog: 1054, 4740, 167, 467, 391, 1701, 731, 1008, 967, 386, 178, 991, 1070, 214, 78144, 168, 
Client: 1.15.0 SoD
Questie: v9.2.13
Locale: enUS

## Screenshots

N/a

## Questie & WoW version

v9.2.13
WoW Classic SoD - v1.15.052409

### PR DESCRIPTION
**Pull request by Sommos**
## Issue references

Fixes #5414

## Proposed changes

- ADDED: 'Alonso the Dragonslayer' is now in the QuestDB, and should be accessible to users.

### CODE REVIEW

File: Database/Corrections/Automatic/sodBaseQuests.lua [hunk 1]
```diff
@@ -470,6 +470,15 @@ function SeasonOfDiscovery:LoadBaseQuests()
...
             [questKeys.requiredClasses] = classIDs.MAGE,
         },
+        [78144] = {
+            [questKeys.name] = "Alonso the Dragonslayer",
+            [questKeys.startedBy] = {{210995}},
+            [questKeys.finishedBy] = {{210995,}},
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 25,
+            [questKeys.requiredRaces] = raceIDs.NONE,
+            [questKeys.requiredClasses] = classIDs.WARRIOR,
```
Comment (BreakBB) [inline Database/Corrections/Automatic/sodBaseQuests.lua:480]:
```suggestion
            [questKeys.requiredClasses] = classIDs.WARRIOR,
            [questKeys.objectivesText] = {"Accompany Alonso to slay the dragon."},
            [questKeys.objectives] = nil,
```

File: Database/Corrections/sodQuestFixes.lua [hunk 2]
```diff
@@ -239,6 +239,11 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.requiredSpell] = -425170,
             [questKeys.zoneOrSort] = sortKeys.MAGE,
         },
+        [78144] = {
+            [questKeys.questLevel] = -1,
+            [questKeys.requiredSpell] = -425444,
+            [questKeys.zoneOrSort] = sortKeys.WARRIOR,
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:245]:
```suggestion
            [questKeys.zoneOrSort] = sortKeys.WARRIOR,
            [questKeys.preQuestGroup] = {78132,78133,78134},
```

All three quests are required before you get this escort quest.

**Review by BreakBB — COMMENTED** [review]
Can you include in this PR to replace the stupid fake quests I falsely added for this rune?

90096 -> https://www.wowhead.com/classic/quest=78144/alonso-the-dragonslayer ✅ 
90097 -> https://www.wowhead.com/classic/quest=78133/dragonslayers-shield
90098 -> https://www.wowhead.com/classic/quest=78134/dragonslayers-lance
90099 -> https://www.wowhead.com/classic/quest=78132/dragonslayers-helm

File: Database/Corrections/sodQuestFixes.lua (hunk 2 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua:245]: Fixed in: https://github.com/Questie/Questie/pull/5416/commits/b251e6416f0fb61b5e42593546b490594eba3c7b

File: Database/Corrections/Automatic/sodBaseQuests.lua (hunk 1 above)
Comment (Sommos) [reply Database/Corrections/Automatic/sodBaseQuests.lua:480]: Fixed in: https://github.com/Questie/Questie/pull/5416/commits/b251e6416f0fb61b5e42593546b490594eba3c7b

**Review by Sommos — COMMENTED** [review]
This should now have all the fake rune quests removed, and the attributes you wrote up split in to the correct `base` & `fixes` locations.

File: Database/Corrections/sodQuestFixes.lua [hunk 3]
```diff
@@ -229,6 +229,24 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.requiredSpell] = -425170,
             [questKeys.zoneOrSort] = sortKeys.MAGE,
         },
+        [78132] = {
+            [questKeys.startedBy] = {nil,{409758}},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:233]:
```suggestion
            [questKeys.objectives] = {nil,nil,{{209872}}},
```

File: Database/Corrections/sodQuestFixes.lua [hunk 4]
```diff
@@ -229,6 +229,24 @@ function SeasonOfDiscovery:LoadQuests()
...
+            [questKeys.objectivesText] = {"Retrieve the Dragonslayer's Helm and bring it to Alonso in Ashenvale."},
+            [questKeys.requiredSpell] = -425444,
+            [questKeys.zoneOrSort] = sortKeys.WARRIOR,
+        },
+        [78133] = {
+            [questKeys.startedBy] = {nil,{409754}},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:239]:
```suggestion
            [questKeys.objectives] = {nil,nil,{{209873}}},
```

File: Database/Corrections/sodQuestFixes.lua [hunk 5]
```diff
@@ -229,6 +229,24 @@ function SeasonOfDiscovery:LoadQuests()
...
+            [questKeys.objectivesText] = {"Retrieve the Dragonslayer's Shield and bring it to Alonso in Ashenvale."},
+            [questKeys.requiredSpell] = -425444,
+            [questKeys.zoneOrSort] = sortKeys.WARRIOR,
+        },
+        [78134] = {
+            [questKeys.startedBy] = {{3180}},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:245]:
```suggestion
            [questKeys.objectives] = {nil,nil,{{209874}}},
```

**Review by BreakBB — COMMENTED** [review]
The `startedBy` was only added to the fake quests, because these quests are never in the quest log and therefore the `objectives` field would never be used.

But since these are real quests, we can (and should) use that field.

File: Database/Corrections/sodQuestFixes.lua (hunk 5 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua]: Fixed in: https://github.com/Questie/Questie/pull/5416/commits/b8b66e1032045f73d133207f1a9eb4ae59612d21

File: Database/Corrections/sodQuestFixes.lua (hunk 3 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua]: Fixed in: https://github.com/Questie/Questie/pull/5416/commits/b8b66e1032045f73d133207f1a9eb4ae59612d21

File: Database/Corrections/sodQuestFixes.lua (hunk 4 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua]: Fixed in: https://github.com/Questie/Questie/pull/5416/commits/b8b66e1032045f73d133207f1a9eb4ae59612d21

**Pull request merged**